### PR TITLE
Feature: add border for barchart

### PR DESCRIPTION
**Pull request by samiazar**
* **IMPROVEMENT** Added `borderSide` property in [BarChartRodData](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/bar_chart.md#BarChartRodData) and [BarChartRodStackItem](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/bar_chart.md#BarChartRodStackItem) to draw strokes around each bar and rod stack items, #714.

### CODE REVIEW

File: lib/src/utils/utils.dart [hunk 1]
```diff
@@ -82,6 +82,25 @@ BorderRadius? normalizeBorderRadius(BorderRadius? borderRadius, double width) {
...
+  }
+
+  double borderWidth;
+  if (borderSide.width > width / 2) {
+    borderWidth = width / 2.toDouble();
+  } else {
+    borderWidth = borderSide.width;
+  }
+
+  return BorderSide(color: borderSide.color, width: borderWidth, style: borderSide.style);
```
Comment (imaNNeoFighT) [inline lib/src/utils/utils.dart:101]: You can use borderSide.copy() here

File: lib/src/chart/bar_chart/bar_chart_painter.dart [hunk 2]
```diff
@@ -317,6 +326,9 @@ class BarChartPainter extends AxisChartPainter<BarChartData> {
               canvasWrapper.drawRRect(barRRect, _barPaint);
               canvasWrapper.restore();
             }
+            // draw border stroke for each stack item
+            _drawRodStacksBorderStroke(
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_painter.dart:330]: Above we iterated over stack items to draw them, why we iterated over them again inside this function?
I think we can use one iteration over stack items at all.

**Comment by imaNNeoFighT** [comment]
Everything looks good to me.
Thanks Sami!

File: lib/src/utils/utils.dart (hunk 1 above)
Comment (samiazar) [reply lib/src/utils/utils.dart:101]: Thank you for the good tip.

File: lib/src/chart/bar_chart/bar_chart_painter.dart (hunk 2 above)
Comment (samiazar) [reply lib/src/chart/bar_chart/bar_chart_painter.dart:330]: You're right, I've changed the code.

**Pull request merged**